docs/mobile-apps/automated-testing/about-appium.md

```diff
@@ -0,0 +1,437 @@
+
+_Implicit waits_ set a maximum time that the Appium server will continue trying to find an element. Using implicit waits is not recommended because application response times are not definitely predictable and fixed elapsed times are not applicable to all interactions. Using explicit waits requires more technical sophistication, but is a Sauce Labs best practice.
+
+This example below illustrates how you could use an implicit wait to anticipate web browser response after submitting the login form:
+
+```java
+driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+```
+
+#### **Explicit Waits Strategy**
+
+_Explicit waits_ wait until an expected condition occurs on the web page, or until a maximum wait time elapses. To use an explicit wait, you create an instance of the WebDriverWait class with a maximum wait time, and you invoke its until method with an expected condition. 
+
+The WebDriver API provides an `ExpectedConditions` class with methods for various standard types of expected condition. These methods return an instance of an expected condition class. You can pass an invocation of these standard expected-condition methods as argument values to `until` method. You can also pass - in ways that your programming language and its WebDriver API support - any function, code block, or closure that returns a boolean value or an object reference to a found web element as an argument value to the until method. How this is done varies over programming languages. The `until` method checks repeatedly, until the maximum wait time elapses, for a true boolean return value or a non-`null` object reference, as an indication that the expected condition has occurred.
+
+This example below illustrates how you could use an explicit wait to anticipate web browser response after submitting the login form.
+
+```java
+import org.openqa.selenium.support.ui.ExpectedConditions; 
+import org.openqa.selenium.support.ui.WebDriverWait; 
+WebDriverWait wait = new WebDriverWait(driver, 10); WebElement messageElement = wait.until( ExpectedConditions.presenceOfElementLocated(By.id("loginResponse")) );
+```
+
+>**NOTE**: Do not mix implicit and explicit waits. Doing so can cause unpredictable wait times. For example setting an implicit wait of 10s and an explicit wait of 15 seconds, could cause a timeout to occur after 20 seconds.
+
+<br/>
+
+### 6. Run Tests and Record Your Results
+
+This is where you log whether your test passed, failed, or returned an error.
+
+Running tests and recording test results is the ultimate purpose of your test script: you run tests in an automated test script in order to evaluate function and performance in the AUT, without requiring human interaction.
+
+#### **Test Frameworks**
+
+To run test and to record test results, you use methods of a test framework for your programming language. There are many available test frameworks, including the frameworks in the so-called XUnit family, which includes:
```

Comment:
Can we also link to the Sauce Bindings for all languages except JS?

Reply:
I didn't make this doc- simply ported over from old appium.md that is currently published here (https://docs.saucelabs.com/mobile-apps/automated-testing/appium)-- feel free to add any copy-editing commits you would like